Shaders/UIMask.fx

```diff
@@ -1,189 +1,266 @@
-      https://msdn.microsoft.com/en-us/library/windows/desktop/dd375731(v=vs.85).aspx
-
-    --To make a custom mask:
-
-      1-Take a screenshot of your game with the HUD enabled,
-       preferrably with any effects disabled for maximum visibility.
-
-      2-Open the screenshot with your preferred image editor program, I use GIMP.
-
-      3-Make a background white layer if there isn't one already.
-        Be sure to leave it behind your actual screenshot for the while.
-
-      4-Make an empty layer for the mask itself, you can call it "mask".
-
-      5-Having selected the mask layer, paint the places where HUD constantly is,
-        such as health bars, important messages, minimaps etc.
-      
-      6-Delete or make your screenshot layer invisible.
-
-      7-Before saving your mask, let's do some gaussian blurring to improve it's look and feel:
-        For every step of blurring you want to do, make a new layer, such as:
-        Mask - Blur16x16
-        Mask - Blur8x8
-        Mask - Blur4x4
-        Mask - Blur2x2
-        Mask - NoBlur
-        You should use your image editor's default gaussian blurring filter, if there is one.
-        This avoids possible artifacts and makes the mask blend more easily on the eyes.
-        You may not need this if your mask is accurate enough and/or the HUD is simple enough.
-
-      8-Now save the final image as "UIMask.png" in your textures folder and you're done!
-
-    
-    MIT Licensed:
-
-    Copyright (c) 2017 Lucas Melo
-
-    Permission is hereby granted, free of charge, to any person obtaining a copy
-    of this software and associated documentation files (the "Software"), to deal
-    in the Software without restriction, including without limitation the rights
-    to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
-    copies of the Software, and to permit persons to whom the Software is
-    furnished to do so, subject to the following conditions:
-
-    The above copyright notice and this permission notice shall be included in all
-    copies or substantial portions of the Software.
-
-    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
-    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
-    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
-    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
-    SOFTWARE.
+	Simple UIMask shader by luluco250
+
+	I have no idea why this was never ported back to ReShade 3.0 from 2.0,
+	but if you missed it, here it is.
+
+	It doesn't feature the auto mask from the original shader.
+
+	It does feature a new multi-channnel masking feature. UI masks can now contain
+	separate 'modes' within each of the three coloror channels.
```

Comment:
`coloror`: Looks like a search + replace gone wrong? There are a couple of these (easy to spot when vieweing the diff with whitespace changes ignored).

Reply:
Oh for sure, I was replacing "col" with "color" in the Main shader, my mistake.